### DISCUSSION TITLE
Compilations and Various Artists albums, some tagged as album artist some as various issue

### DISCUSSION THREAD

**Discussion by rortonuk** [beetbox/beets#5448] [Q&A]
Working with Beets to sort all my 'Artist' based Albums, and thats working well. 

Problem I have is that I have hundreds and hundreds of 'Compilation Albums' - DJ Mix CDs and such like, and some of them are detected as an Album Artist (assume due to Music Brainz having this set), so for example Compilation Album "Essential Selection" has an Album Artist as "Pete Tong" 

I don't want to tag my compilation albums this way, some I will have no idea who was the album artist, so I feel I should have all my compilations/various artists albums in a 'Various Artists' directory.

How do I achieve this? Its like I need to overwrite what MusicBrainz is saying about the album artist and populate the album artists filed with 'Various Artists'

Thanks

**Comment by rortonuk** [comment]
forgot to mention, I have beets running in docker, so I got a 'production' docker with all my Artists albums in which im happy with, and then a test docker im using to get the various artists correct so I can mess about with this to my hearts content and not affect the main instance.

**Comment by jackwilsdon** [comment]
You should be able to set `comp` on the album with `beet modify comp=True album:"Essential Selection"` (it isn't marked as a compilation by beets because the album artist isn't https://musicbrainz.org/artist/89ad4ac3-39f7-470e-963a-56509c546377, but maybe we should also be checking album type as that indicates it's a compilation: https://musicbrainz.org/release/18d21bc6-3de2-48e4-b6a4-1a2d4c3084c3).

This will move it to your configured compilations path (which you can name "Various Artists" in your configuration if you want): https://github.com/beetbox/beets/blob/04ee04150ac6ce41393c4e8e0d04030f3c5e2596/beets/config_default.yaml#L101-L104

**Comment by rortonuk** [comment]
hi, thanks for the reply. That means I have to import all these hundreds of Albums first, then write a script to extract all their names and then set the flag to true? wondered if there was an easier way of doing this before I bring the albums into beets?

**Reply by jackwilsdon** [reply]
I don't think beets uses any of your existing tags to set `comp` - it's strictly from MusicBrainz (assuming you are using MB):

https://github.com/beetbox/beets/blob/03f1205629ad17f123a190040361babc41c18afc/beets/autotag/mb.py#L537

https://github.com/beetbox/beets/blob/03f1205629ad17f123a190040361babc41c18afc/beets/autotag/__init__.py#L252-L253

You could try setting it during import if you're _only_ importing compilations using `beet import --set comp=True ...`, although I'm not 100% sure if that will work.
